### PR TITLE
Add tests for scenario 14.2

### PR DESCRIPTION
**Pull request by sdkottegoda**
## Purpose
Adds tests for scenario 14.2 - Scheduling a task by executing a java task
Also contains some refactoring to use common methods.

### CODE REVIEW

File: product-scenarios/14-periodically-execute-an-integration-process/14.2-scheduling-a-customized-java-task/14.2.2-scheduling-a-java-task-with-cron/src/test/resources/keystores/key.crt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+-----BEGIN CERTIFICATE-----
```
Comment (dulanjalidilmi) [inline product-scenarios/14-periodically-execute-an-integration-process/14.2-scheduling-a-customized-java-task/14.2.2-scheduling-a-java-task-with-cron/src/test/resources/keystores/key.crt]: Shall we maintain keystores in scenario level?

File: product-scenarios/14-periodically-execute-an-integration-process/14.2-scheduling-a-customized-java-task/14.2.1-scheduling-a-java-task-with-interval/src/test/resources/keystores/key.crt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+-----BEGIN CERTIFICATE-----
```
Comment (dulanjalidilmi) [inline product-scenarios/14-periodically-execute-an-integration-process/14.2-scheduling-a-customized-java-task/14.2.1-scheduling-a-java-task-with-interval/src/test/resources/keystores/key.crt]: Shall we maintain keystores in scenario level?

File: product-scenarios/14-periodically-execute-an-integration-process/14.2-scheduling-a-customized-java-task/14.2.2-scheduling-a-java-task-with-cron/src/test/resources/keystores/key.crt (hunk 1 above)
Comment (sdkottegoda) [reply product-scenarios/14-periodically-execute-an-integration-process/14.2-scheduling-a-customized-java-task/14.2.2-scheduling-a-java-task-with-cron/src/test/resources/keystores/key.crt]: done

File: product-scenarios/14-periodically-execute-an-integration-process/14.2-scheduling-a-customized-java-task/14.2.1-scheduling-a-java-task-with-interval/src/test/resources/keystores/key.crt (hunk 2 above)
Comment (sdkottegoda) [reply product-scenarios/14-periodically-execute-an-integration-process/14.2-scheduling-a-customized-java-task/14.2.1-scheduling-a-java-task-with-interval/src/test/resources/keystores/key.crt]: done

File: product-scenarios/14-periodically-execute-an-integration-process/14-test-commons/src/main/java/org/wso2/carbon/ei/scenario/test/Scenario14TestBase.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import org.testng.Assert;
+import org.testng.SkipException;
+import org.wso2.carbon.esb.scenario.test.common.ScenarioConstants;
+import org.wso2.carbon.esb.scenario.test.common.ScenarioTestBase;
+import org.wso2.carbon.esb.scenario.test.common.http.HTTPUtils;
+import org.wso2.carbon.esb.scenario.test.common.http.RESTClient;
+
+import java.io.IOException;
+
+public class Scenario14TestBase extends ScenarioTestBase {
```
Comment (milindaperera) [inline product-scenarios/14-periodically-execute-an-integration-process/14-test-commons/src/main/java/org/wso2/carbon/ei/scenario/test/Scenario14TestBase.java]: shall we add class commet

Comment (sdkottegoda) [reply product-scenarios/14-periodically-execute-an-integration-process/14-test-commons/src/main/java/org/wso2/carbon/ei/scenario/test/Scenario14TestBase.java]: done

**Pull request merged**